core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/GroupIterator.java

```diff
@@ -106,16 +104,17 @@ public GroupIterator(EvaluationStrategy strategy, Group group, BindingSet parent
 
 		if (this.iterationCacheSyncThreshold > 0) {
 			try {
-				this.tempFile = File.createTempFile("group-eval", null);
+				this.db = DBMaker
+						.newFileDB(File.createTempFile("group-eval", null))
+						.deleteFilesAfterClose()
+						.closeOnJvmShutdown()
+						.make();
 			} catch (IOException e) {
 				throw new QueryEvaluationException("could not initialize temp db", e);
 			}
-			this.db = DBMaker.newFileDB(tempFile).deleteFilesAfterClose().closeOnJvmShutdown().make();
 		} else {
-			this.tempFile = null;
 			this.db = null;
 		}
-		super.setIterator(createIterator());
```

Comment:
This was the issue @JervenBolleman @kenwenzel

Reply:
There was a call to `createIterator()` inside the constructor of the GroupedIterator which wasn't there before. When I removed it performance improves.

Reply:
Ah @hmottestad that is definitely a bug that snuck in during rebasing.